webapp/src/tutorial.tsx

```diff
@@ -16,6 +16,21 @@ import * as editortoolbar from "./editortoolbar";
 
 type ISettingsProps = pxt.editor.ISettingsProps;
 
+export function xmlUpgrades(code: string[], language?: string,) {
+    compiler.getBlocksAsync()
```

Comment:
maybe wrap the entire thing in `if (pxt.editor.upgradeXml)` instead of just the bit in the last `then` to avoid unnecessary work?

Reply:
:O yes I will do that tytyty